helm_charts_updater/git.py

```
@@ -58,37 +145,72 @@ def push_changes(self, chart_version, app_name: str, version: str, old_version:
 
-        for chart in Path(self.clone_path).rglob("Chart.yaml"):
-            # We want to avoid including dependencies
-            # in the resulting Charts list
-            if len(str(chart).split("/")) <= 4:
-                with open(chart, "r") as file:
+        for chart_path in Path(self.clone_path).rglob("Chart.yaml"):
+            # Filter out dependency charts by checking path depth
+            # Top-level charts: clone_path/charts_path/chart_name/Chart.yaml (4 parts)
+            # Dependencies: clone_path/charts_path/chart_name/charts/dep/Chart.yaml (6+ parts)
+            if len(chart_path.parts) <= 4:
+                with open(chart_path, "r", encoding="utf-8") as file:
```

Ruff (0.14.13)

Unnecessary mode argument

Remove mode argument

(UP015)